### DISCUSSION TITLE
Question on how to add the coefficient of profit for each pickup and delivery pair

### DISCUSSION THREAD

**Discussion by lewis-xuwei** [google/or-tools#3367] [Routing (and legacy CP) questions]
For each PD pair, it has a standalone profit value. We want to maximize the total profits.

**Comment by Mizux** [comment]
recall me the CP-SAT prize collecting model: https://github.com/google/or-tools/blob/main/examples/python/prize_collecting_tsp_sat.py
pro: be able to easily model profit maximization
cons: it is not a "routing customised solver" so you won't have all guided local search and first solution heuristic specialised for a routing VRP problem.

If using routing since the objective is hard coded to minimize the sum of all costs so If I were you I would add a disjunction penalty high enough to be the dominant factor (i.e. one order of magnitude higher than the sum of arc cost so solver has incentive to first try to optimize profit then optimize the route) and whose value is proportional to the profit aka higher profit == higher penalty so solver should maximise profit while trying to minimize the sum of penalty cost
